state/watcher/hubwatcher.go

```diff
@@ -300,13 +411,61 @@ func (w *HubWatcher) handle(req interface{}) {
 				e.ch = nil
 			}
 		}
+	case reqStats:
+		var watchCount uint64
+		for _, watches := range w.watches {
+			watchCount += uint64(len(watches))
+		}
+		var revnoMapBytes uintptr
+		// TODO: (jam) 2018-11-05 there would be ways to be more accurate about the sizes,
+		// but this is a rough approximation. We should also include the size overhead of all the objects.
+		// watchKey, str pointer and size overhead, map etc.
+		revnoMapBytes += reflect.TypeOf(w.current).Size()
+		for wKey, _ := range w.current {
```

Comment:
Its approximate enough that I'd be willing to drop it entirely, though one thing I've been uncovering at least in testing so far is that we may not need '.current' at all.

Specifically, it explicitly handles the case where you read a document, and *then* start watching it, rather than starting a watch and using that watch to give you the document.
We've generally pushed towards the latter form, and if we enforced that, we could get rid of all the caching we do in current in all the hub watchers.

Anyway, withing getting side tracked, we can move this to be updated on the fly. It means doing a double lookup on a Change to check if the object already existed, but with any luck the compiler can just inline it.